assert-plus/index.d.ts

```diff
@@ -3,65 +3,505 @@
+ * 
+ * ```js
+ * // WARNING: This does not throw an AssertionError!
+ * assert.deepEqual(Error('a'), Error('b'));
+ * ```
+ * 
+ * "Deep" equality means that the enumerable "own" properties of child objects are evaluated also.
+ * 
+ * If the values are not equal, an `AssertionError` is thrown with a `message` property
+ * set equal to the value of the `message` parameter. If the `message` parameter is undefined,
+ * a default error message is assigned.
+ */
 export function deepEqual(actual: any, expected: any, message ?: string): void;
+
+/**
+ * Tests for any deep inequality. Opposite of `assert.deepEqual()`.
+ * 
+ * ```js
+ * const assert = require('assert');
+ * 
+ * const obj1 = { a : { b : 1 } };
+ * const obj2 = { a : { b : 2 } };
+ * const obj3 = { a : { b : 1 } };
+ * const obj4 = Object.create(obj1);
+ * 
+ * assert.notDeepEqual(obj1, obj1);
+ * // AssertionError: { a: { b: 1 } } notDeepEqual { a: { b: 1 } }
+ * 
+ * assert.notDeepEqual(obj1, obj2);
+ * // OK, obj1 and obj2 are not deeply equal
+ * 
+ * assert.notDeepEqual(obj1, obj3);
+ * // AssertionError: { a: { b: 1 } } notDeepEqual { a: { b: 1 } }
+ * 
+ * assert.notDeepEqual(obj1, obj4);
+ * // OK, obj1 and obj2 are not deeply equal
+ * ```
+ * 
+ * If the values are deeply equal, an `AssertionError` is thrown with
+ * a `message` property set equal to the value of the `message` parameter.
+ * If the `message` parameter is undefined, a default error message is assigned.
+ */
 export function notDeepEqual(actual: any, expected: any, message ?: string): void;
+
+/**
+ * Tests strict equality as determined by the strict equality operator ( `===` ).
+ * 
+ * ```js
+ * const assert = require('assert');
+ * 
+ * assert.strictEqual(1, 2);
+ * // AssertionError: 1 === 2
+ * 
+ * assert.strictEqual(1, 1);
+ * // OK
+ * 
+ * assert.strictEqual(1, '1');
+ * // AssertionError: 1 === '1'
+ * ```
+ * 
+ * If the values are not strictly equal, an `AssertionError` is thrown with
+ * a `message` property set equal to the value of the `message` parameter.
+ * If the `message` parameter is undefined, a default error message is assigned.
+ */
 export function strictEqual(actual: any, expected: any, message ?: string): void;
+
+/**
+ * Tests strict inequality as determined by the strict not equal operator ( `!==` ).
+ * 
+ * ```js
+ * const assert = require('assert');
+ * 
+ * assert.notStrictEqual(1, 2);
+ * // OK
+ * 
+ * assert.notStrictEqual(1, 1);
+ * // AssertionError: 1 !== 1
+ * 
+ * assert.notStrictEqual(1, '1');
+ * // OK
+ * ```
+ * 
+ * If the values are strictly equal, an `AssertionError` is thrown with a `message` property
+ * set equal to the value of the `message` parameter. If the `message` parameter is undefined,
+ * a default error message is assigned.
+ */
 export function notStrictEqual(actual: any, expected: any, message ?: string): void;
+
+/**
+ * 
+ */
 export function throws(block: any, error ?: any, message ?: string): void;
```

Comment:
Because the world is not ideal. For example, why just not use generics [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/bc1da27217d7b57dd43f7f2c182c4ff89bf51392/node/index.d.ts#L3611)?

```ts
export function strictEqual(actual: any, expected: any, message?: string): void;
```